scripts/SMD_chip_package_rlc-etc/size_definitions/size_capacitor_chip.yaml

```diff
@@ -0,0 +1,143 @@
+C_0504:
+    code_imperial: "0504"
+    code_metric: "1310"
+    body_length:
+        minimum: 1.02
+        maximum: 1.32
+    body_width:
+        minimum: 0.77
+        maximum: 1.27
+    terminal_length:
+        minimum: 0.13
+        maximum: 0.38
+    size_info: '(Body size source: IPC-SM-782 page 76, https://www.pcb-3d.com/wordpress/wp-content/uploads/ipc-sm-782a_amendment_1_and_2.pdf)'
+
+C_0603:
+    code_imperial: "0603"
+    code_metric: "1608"
+    body_length:
+        minimum: 1.45
+        maximum: 1.75
+    body_width:
+        minimum: 0.65
+        maximum: 0.95
+    terminal_length:
+        minimum: 0.2
+        maximum: 0.5
+    size_info: '(Body size source: IPC-SM-782 page 76, https://www.pcb-3d.com/wordpress/wp-content/uploads/ipc-sm-782a_amendment_1_and_2.pdf)'
+
+C_0805:
+    code_imperial: "0805"
+    code_metric: "2012"
+    body_length:
+        minimum: 1.8
+        maximum: 2.2
+    body_width:
+        minimum: 1.05
+        maximum: 1.45
+    terminal_length:
+        minimum: 0.25
+        maximum: 0.55 # only deviation from IPC is reducing this from 0.75mm
```

Comment:
Why that deviation? (IPC has 0.75)

Reply:
It's been quite a while, but as best I can recall it was to maintain the footprints more close to the existing ones. Rene had compiled a list of Cs when the R body dimensions were being used and there were issues. See https://github.com/KiCad/kicad-footprints/issues/711 and https://github.com/KiCad/kicad-footprints/pull/712. I noted this above so Rene may give some feedback, but at this point if you want to go strictly with IPC that's fine with me. In fact, I like it even better but we should be prepared for any comments about it here or on the forum. So we will need to evaluate the change in pad size/location before deciding, I think. Want me to take a look at that or do you want to?

Reply:
Currently, our C 0805 (non-handsolder) footprint is:
- Pad centers: +/-0.9375mm
- Pad width: 0.975mm
- Pad inside: 0.9mm
- Pad outside: 2.85mm

The script currently results in:
- Pad centers: +/-0.95mm
- Pad width: 1.0mm
- Pad inside: 0.9mm
- Pad outside: 2.9mm

If I use the 0.75mm terminal length from IPC I get this:
- Pad centers: +/-0.875mm
- Pad width: 1.15mm
- Pad inside: 0.6mm
- Pad outside: 2.9mm

So we kinda end up back where we were when Wayne filed the issue I linked above. The pad spacing is quite small. The adjustment I made here was to roughly match the existing 0805 cap footprint.

We could enter the IPC-SM-782 terminal spacing instead of terminal width, and then we get this:
- Pad centers: +/-0.8mm
- Pad width: 1.3mm
- Pad inside: 0.3mm
- Pad outside: 2.9mm

So even worse, at least from the original comment made by Wayne. My solution was to align with the IPC numbers for everything but this dimension, which happily resulted in good matching with what Rene pulled out of datasheets once the proper capacitor body dimension info was used.

Do we have any other reference? Something to check our dimensions against? Another IPC-compliant footprint generator? Wayne's concern seemed valid to me and once Rene got actual part dimensions this seemed reasonable. I wouldn't object to making further changes, but without a reason I'm somewhat resistant to matching the IPC numbers here and going against what feels like the right approach (as opposed to every other situation involving our IPC generators where I saw go with it and that's how it should be!).

Thoughts?

Reply:
I opened my old Eagle 6.3 to see what is in there.
* Pad centers +/- 0.95mm
* Pad width: 1.5mm
* Pad inside: 0.6mm
* Pad outside: 3.2mm

Some more stuff I researched:
* Kemet Datasheets have exactly the same dimensions and tolerances as IPC-SM-782
* Yaego Datasheets have exactly the same dimensions and tolerances as IPC-SM-782. But they also sell caps with tighter tolerances where the spacing between the pads could be considerably larger

Yaego also has a nice table for footprints:
![image](https://user-images.githubusercontent.com/17516572/93255403-2a01b280-f79a-11ea-9f74-a9911ad4b5fc.png)

Following their recommendation (Nominal Density) we would get:
* Pad centers +/- 0.9mm
* Pad width: 1.15mm
* Pad inside: 0.325mm
* Pad outside: 2.95mm

Reply:
Good info. Thanks! Do you have links? Especially for the tighter-tolerance part. (I'm just lazy but I should search myself.)

The footprint in that screenshot looks like quite close to the IPC one, then.  And they only note IPC-7351, not IPC-73651B, so that could account for the slight difference. The fillet goals do tend to change a bit as the doc was updated.

Here is an image, top to bottom, of 0603, 0805, 0805 using strictly IPC, and 1206:
![image](https://user-images.githubusercontent.com/1936989/93258582-65c55980-f753-11ea-80d5-980b2079dfe4.png)

I'm still a bit concerned. Wayne's comments I linked above about the 0805 cap footprint still apply, even with the corrected dimensions. And especially since there are caps in the market with tighter tolerances. Should we have multiple 0805 cap footprints? Or adjust this one dimension from IPC as I originally was thinking? Something else?

Reply:
https://www.yageo.com/upload/media/product/productsearch/datasheet/mlcc/UPY-GPHC_X7R_6.3V-to-50V_18.pdf
![image](https://user-images.githubusercontent.com/17516572/93262895-41926880-f7a5-11ea-95f2-ffe9771e23e6.png)

Reply:
So the 0805 body size (length and width) tolerance drops from +/-0.2mm to +/-0.1mm, but the terminal length stays the same. (And just for kicks, the 0402 shows a terminal length of 0.15-0.35mm. IPC shows 0.1-0.3mm. Not tighter, just shifted.)

If I plug that in, using 0.75mm max terminal length per IPC, I get:
Pad centers: +/-0.8625mm
Pad width: 1.125mm
Pad inside: 0.6mm
Pad outside: 2.85mm

Not any substantial change.

I'm still uncomfortable with such closely-spaced pads. I prefer my slight adjustment for max terminal length from 0.75mm to 0.55mm. IMHO this makes our library most compatible for the most users in the most circumstances.

Reply:
Agree, we should hack/change that footprint to get more spacing between the pads.
0.55mm seems like a good choice.